compiler/scopelang/from_desugared.ml

```diff
@@ -65,36 +66,53 @@ let rec translate_expr (ctx : ctx) (e : Desugared.Ast.expr) :
     Expr.elocation
       (ScopelangScopeVar
          (match
-            ScopeVarMap.find (Marked.unmark s_var) ctx.scope_var_mapping
+            ScopeVar.Map.find (Marked.unmark s_var) ctx.scope_var_mapping
           with
          | WholeVar _ -> failwith "should not happen"
          | States states -> Marked.same_mark_as (List.assoc state states) s_var))
       m
   | EVar v -> Expr.evar (Var.Map.find v ctx.var_mapping) m
   | EStruct { name; fields } ->
-    Expr.estruct name (StructFieldMap.map (translate_expr ctx) fields) m
-  | EStructAccess { e; field; name } ->
-    Expr.estructaccess (translate_expr ctx e) field name m
+    Expr.estruct name (StructField.Map.map (translate_expr ctx) fields) m
+  | EDStructAccess { name_opt = None; _ } ->
+    (* Note: this could only happen if disambiguation was disabled. If we want
+       to support it, we should still allow this case when the field has only
+       one possible matching structure *)
+    Errors.raise_spanned_error (Expr.mark_pos m)
+      "Ambiguous structure field access"
```

Comment:
The error message should contain a suggestion about how to disambiguate. Ideally with the names of all the structs corresponding to the field access.

Reply:
This case will never be reached unless disambiguation is disabled though (is that something we might want to support ? I left that door open but don't know what the use-cases would be)
Normally, after disambiguation, `name_opt` will always be `Some`, or an error would already have been raised if the struct type couldn't be resolved (that should never happen on a well-typed program).

Reply:
Aren't there cases where disambiguation can fail ? We might have cases later when the language evolves where we can't guarantee desambiguation will always work

Reply:
I... don't think there should be, but that warrants some more formalization :innocent: 

At the moment, anyway, there are none — and by "leaving the door open" I meant that I always kept the ways to manually disambiguate programs.

I guess the future extensions you have in mind would include the ability to define polymorphic functions. If it's real polymorphism, you won't be able to use field accesses on that type anyway so we're fine (the error would be triggered earlier than here, during the disambiguation pass) ;

The only case I could see where this would become problematic is if we start to want type inference on variables, e.g. expecting `fun x -> x.foo` to infer that `x` is `some_struct`. Even if we allow such functions, I would be in favour of asking the user to disambiguate at the point of introduction of the variable when needed.

Reply:
A more straightforward answer: yes, it can fail, but that would happen in the disambiguation pass, not here.

Currently, failure would only happen on badly typed programs. But indeed, with language extensions, we can imagine cases where it would fail on a a correct program and require manual disambiguation.

Reply:
Ok some we can leave the message as it is now, and make it better if ever it's really used later.